src/sdk/primitive.hpp

```diff
@@ -0,0 +1,152 @@
+#ifndef primitive_hpp
+#define primitive_hpp
+
+#include "accessor.hpp"
+#include "type_info.hpp"
+#include "property.hpp"
+
+namespace realm::sdk {
+
+// MARK: Primitive Accessor
+template <typename T, typename E>
+struct Accessor : public AccessorBase {
+    using base = AccessorBase;
+    using base::base;
+    using type_info = TypeInfo<T>;
+
+    Accessor(const AccessorBase& o) : base(o)
+    {
+    }
+
+    typename type_info::realm_type get()
+    {
+        return base::get_obj().template get<typename type_info::realm_type>(base::get_column_key());
+    }
+
+    typename type_info::const_realm_type get() const
+    {
+        return base::get_obj().template get<typename type_info::const_realm_type>(base::get_column_key());
+    }
+
+    void set(const T& value)
+    {
+        base::get_obj().template set<typename type_info::realm_type>(base::get_column_key(),
+                                                                     static_cast<typename type_info::realm_type>(value));
+    }
+
+    T to_value_type(const typename type_info::const_realm_type& realm_value) const
+    {
+        return static_cast<T>(realm_value);
+    }
+
+    typename type_info::realm_type to_realm_type(const T& value) const
+    {
+        return static_cast<typename type_info::realm_type>(value);
+    }
+};
+
+template <typename T>
+using PrimitiveAccessor = Accessor<T>;
+
+/**
+ MARK: PrimitiveBase
+ Base `Property` specialization for any primitive type. "Primitive"
+ is defined here as any non collection or polymorphic type, such as
+ int, long, ObjectId, time_point, std::string, etc..
+ */
+template <typename Impl, typename T>
+struct PrimitiveBase : PropertyBase
+{
+    using value_type = T;
+    using impl = Impl;
+    using PropertyBase::PropertyBase;
```

Comment:
Do you really want any `PrimitiveBase` to be constructible from any `const PropertyBase&`?